pkg/homedir/homedir.go

```diff
@@ -50,3 +36,47 @@ func GetCacheHome() (string, error) {
 	}
 	return filepath.Join(home, ".cache"), nil
 }
+
+// SetXdgDirs ensures the XDG_RUNTIME_DIR env and XDG_CONFIG_HOME variables are set.
+// containers/image uses XDG_RUNTIME_DIR to locate the auth file, XDG_CONFIG_HOME is
+// use for the containers.conf configuration file.
+func SetXdgDirs(rootless bool, rootlessUID int) error {
+	if !rootless {
+		os.Unsetenv("XDG_RUNTIME_DIR")
+		os.Unsetenv("XDG_CONFIG_HOME")
```

Comment:
Yes, the Jira card says “Root should ignore XDG_RUNTIME_DIR and unset it such that other tools (e.g., crun) don't use it either.”. but it provides absolutely no explanation.

Actively interfering with users’ configuration seems unexpected. Please write down in detail what’s the problem and solution, so that the code can be maintained over time.